doc/gearbox/gearbox_mdio-HLD.md

```diff
@@ -0,0 +1,246 @@
+		    
+		    for (auto i = 0; i < attr_count; i++)
+		    {
+		        if (attr_list[i].id == SAI_SWITCH_ATTR_REGISTER_READ && m_context !=0)
+		        {
+		        attr.id = SAI_SWITCH_ATTR_REGISTER_READ;
+		        attr.value.ptr =  (void *) m_mdio_read;
+		        attr_copy.push_back(attr);
+		        }
+		        else if (attr_list[i].id == SAI_SWITCH_ATTR_REGISTER_WRITE)
+		        {
+		        attr.id = SAI_SWITCH_ATTR_REGISTER_WRITE;
+		        attr.value.ptr =  (void *) m_mdio_write ;
+		        attr_copy.push_back(attr);
+		        }
+		        else
+		        {
+		        attr_copy.push_back(attr_list[i]);
+		        }
+		    }
+		    sai_attribute_t *attr_array = attr_copy.data();
+		    auto status = info->create(&mk, switchId, attr_count, attr_array);
+		    ...
+		}
+		else
+		{
+		    auto status = info->create(&mk, switchId, attr_count, attr_list);
+		    ...
+		}
+	...
+	}
+
+### SAI API 
+
+When a switch platform with external PHY is connected to the MDIO bus from the switch NPU, the external PHY is accessed through the SAI and syncd. The existing SAI switch api already has mdio access functions switch\_mdio\_read and switch\_mdio\_write defined, but it does not distinguish the Clause 45 mdio from the Clause 22 mdio. New mdio access functions switch\_mdio\_cl22\_read and switch\_mdio\_cl22\_write are added to the SAI switch api to handle the Clause 22 mdio only.
+
+	typedef struct _sai_switch_api_t
+	{
+	    sai_create_switch_fn                   create_switch;
+	    sai_remove_switch_fn                   remove_switch;
+		...
+	    sai_switch_mdio_read_fn                switch_mdio_read;
+	    sai_switch_mdio_write_fn               switch_mdio_write;
+		...
+	+   sai_switch_mdio_cl22_read_fn           switch_mdio_cl22_read;
+	+   sai_switch_mdio_cl22_write_fn          switch_mdio_cl22_write;
+	} sai_switch_api_t;
+
+For backward compatibility of the SAI API before above change, the Clause 22 mdio device can still be accessed using the existing mdio access functions together with extra configuration in NPU SAI/SDK or experimental switch attribute. For example, new SAI soc property is added to the config.bcm file for a switch platform with external PHY using the Clause 22 mdio registers:
+
+	device/arista/x86_64-arista_720dt_48s/td3x2-a720dt-48s-flex.config.bcm:
+	...
+	+sai_mdio_access_clause22=1
+
+### Configuration and management 
+Each device with external PHY should have a configuration file gearbox_config.json. One JSON key "lib\_name" has the value of the PAI library name. We add 2 more JSON keys. One new JSON key "phy\_access\_lib\_name" has the value of the mdio access library name. The other new JSON key "mdio\_cl22\_only" indicates the PHY device using clause 22 mdio.
+
+gearbox_config.json:
+
+	"phys": [
+	{   
+	  "phy_id": 1,
+	  ...
+	  "lib_name": "/usr/lib/libpai_layer.so",
+	  ...
+	  "phy_access": "mdio",
+	+ "phy_access_lib_name": "/usr/lib/libgbsyncdaccess.so",
+	+ "mdio_cl22_only": true,
```

Comment:
Instead of a boolean key, I prefer to explicitly specify mdio access method, like

`phy_access_sw ::= "ipc_mdio_cl22" | "ipc_mdio_cl45"`

Reply:
Because MDIO clause 22 could access clause 45 register but not vice versa. The MDIO with clause 22 register is only referred as clause 22 access only.